### PR TITLE
handle uppercase hostname and preferred subnet changes

### PR DESCRIPTION
**Pull request by sadysnaat**

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/76#" title="Author self-approved">sadysnaat</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jzhoucliqr**
You can assign the PR to them by writing `/assign @jzhoucliqr` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/spectro-release-3.4/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: controllers/maascluster_controller.go [hunk 1]
```diff
@@ -216,15 +222,38 @@ func IsRunning(m *infrav1beta1.MaasMachine) bool {
...
 
+func addressInPreferredSubnets(log logr.Logger, preferredSubnets []string, address string) bool {
+	if len(preferredSubnets) == 0 {
+		return true
+	}
+
+	for _, subnet := range preferredSubnets {
+		_, subnetParsed, err := net.ParseCIDR(subnet)
+		if err != nil {
+			log.Error(err, "unable to parse subent", "subnet", subnet)
```
Comment (AmitSahastra) [inline controllers/maascluster_controller.go:242]: * subnet

File: pkg/maas/scope/cluster.go [hunk 2]
```diff
@@ -171,6 +173,32 @@ func (s *ClusterScope) GetClusterMaasMachines() ([]*infrav1beta1.MaasMachine, er
 	return machines, nil
 }
 
+const (
+	maasPreferredSubnetConfigmap = "maas-preferred-subnet"
+	preferredSubnetKey           = "preferredSubnets"
+)
+
+func (s *ClusterScope) GetPreferredSubnets() ([]string, error) {
+	maasPreferredSubent := &v1.ConfigMap{}
```
Comment (AmitSahastra) [inline pkg/maas/scope/cluster.go:182]: maasPreferredSubnet*

**Review by AmitSahastra — APPROVED** [review]
Approved. minor spell correction required.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/76#pullrequestreview-1430415312" title="Approved">AmitSahastra</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/76#" title="Author self-approved">sadysnaat</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jzhoucliqr**
You can assign the PR to them by writing `/assign @jzhoucliqr` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/spectro-release-3.4/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**